modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1388,7 +1593,27 @@ def filter_full_axis(self, axis, func):
             self.dtypes if axis == 0 else None,
         )
 
-    def explode(self, axis, func):
+    def filter_by_types(self, types: List[Hashable]) -> "PandasDataframe":
+        """
+        Allow the user to specify a type or set of types by which to filter the columns.
+
+        Parameters
+        ----------
+        types : list
+            The types to filter columns by.
+
+        Returns
+        -------
+        PandasDataframe
+             A new PandasDataframe from the filter provided.
+        """
+        return self.mask(
+            col_positions=[
+                i for i in range(len(self.columns)) if self.dtypes[i] in types
+            ]
+        )
+
+    def explode(self, axis: int, func: Callable) -> "PandasDataframe":
```

Comment:
Signature of `ModinDataframe.explode` is different from this. Is this expected?

Reply:
Yes - there are some discrepancies because `explode` was implemented before we formalized this algebra. We'll be fixing these discrepancies once this PR lands!